Formula/libretls.rb

```diff
@@ -0,0 +1,32 @@
+class Libretls < Formula
+  desc "Libtls for OpenSSL"
+  homepage "https://git.causal.agency/libretls/about/"
+  url "https://causal.agency/libretls/libretls-3.5.2.tar.gz"
+  sha256 "59ce9961cb1b1a2859cacb9863eeccc3bbeadf014840a1c61a0ac12ad31bcc9e"
+  license "ISC"
```

Comment:
Upstream mentions "Some compat sources are under the 3-clause BSD license or the MIT license."; however, the files based on Debian https://metadata.ftp-master.debian.org/changelogs//main/libr/libretls/libretls_3.5.2-2_copyright are:
- MIT/Expat - `compat/timegm.c` - macOS and Linux both had `-DHAVE_TIMEGM=1`
- BSD-3 - `compat/strcasecmp.c` - I think this should be in `strings.h` or corresponding library?